src/traces/parcoords/parcoords.js

```diff
@@ -121,8 +121,14 @@ function model(layout, d, i) {
         dimensions = trace.dimensions,
         width = layout.width;
 
+    var cs = !!trace.line.colorscale && Lib.isArray(trace.line.color);
+    var lineColor = cs ?
+        trace.line.color :
+        Array.apply(0, Array(trace.dimensions.reduce(function(p, n) {return Math.max(p, n.values.length);}, 0))).map(function() {return 0.5;});
+    var lineColorScale = cs ? trace.line.colorscale : [[0, trace.line.color], [1, trace.line.color]];
+
```

Comment:
In the original code, `calc.js` actually persisted the result of these calculations, see above, these lines:
```
-    trace.line.color = color;
-    trace.line.colorscale = cscale;
```
... so instead of this, I'd just _return_ these two derived things (whereas now, `calc.js` returns `{}`)
Is it how you'd prefer it? I'm swamped with preparation so I'd avoid dead-ends and you may already have a preferred way in mind (or see issues with it).